storagenode/trust/service.go

```diff
@@ -127,18 +127,26 @@ func (pool *Pool) GetSignee(ctx context.Context, id storj.NodeID) (_ signing.Sig
 		}
 	}
 
-	info.mu.Lock()
-	defer info.mu.Unlock()
+	info.mu.RLock()
+	identity := info.identity
+	info.mu.RUnlock()
 
-	if info.identity == nil {
-		identity, err := pool.kademlia.FetchPeerIdentity(ctx, id)
-		if err != nil {
-			return nil, Error.Wrap(err)
+	if identity == nil {
+		info.mu.Lock()
+		defer info.mu.Unlock()
+
+		// Check to see if another goroutine has won
+		identity = info.identity
+		if identity == nil {
+			identity, err = pool.kademlia.FetchPeerIdentity(ctx, id)
+			if err != nil {
+				return nil, Error.Wrap(err)
+			}
+			info.identity = identity
```

Comment:
Shouldnt the look around the check here be fine?
A la
```
identity, err = pool.kademlia.FetchPeerIdentity(ctx, id)
if info.identity == nil {
    info.mu.Lock()
    info.identity = identity
    info.mu.Unlock()
}
```

Reply:
I mean we are running an additional request into the kad network, but that shouldnt be much of an issue right?

Reply:
It would be running potentially N requests if there were N concurrent misses.